src/commands/positive.ts

```diff
@@ -65,184 +54,119 @@ export default class Positive extends ApiKeyCommand {
-            );
-            this.operationIdToResponseAndValidation[operationId][
-              parameterGroupName
-            ].validationFailures = failures;
-          });
-      },
-    );
+  validateResponses = (responses: OperationResponse): void => {
+    for (const operation of this.operations) {
+      for (const exampleGroup of operation.getExampleGroups()) {
+        const operationExampleId = `${operation.getOperationId()}:${exampleGroup.getName()}`;
+
+        const response = responses[operationExampleId];
+        if (response) {
+          if (response.ok) {
+            const validator = new ResponseValidator(operation, response);
+            validator.validate();
+
+            this.operationFailures[
+              operationExampleId
+            ] = validator.getFailures();
+          } else {
+            this.operationFailures[operationExampleId] = [
+              new ValidationFailure(
+                'Response status code was a non 2XX value',
+                [],
+              ),
+            ];
+          }
+        }
+      }
+    }
   };
 
-  executeRequests = (): Promise<void>[] =>
-    this.operationIds
-      .filter((operationId) => {
-        // filter out all the requests that failed parameter validation
-        const existingValidations = this.operationIdToResponseAndValidation[
-          operationId
-        ];
-
-        return this.failedValidationGuard(existingValidations);
-      })
-      .flatMap((operationId) => {
-        const operationParameters = this.operationIdToParameters[operationId];
-
-        // If multiple parameter sets are present (due to example groups), execute once for each
-        if (Array.isArray(operationParameters)) {
-          return operationParameters
-            .filter((parameterExamples) => {
-              const groupName = Object.keys(parameterExamples)[0];
-              const existingValidations = this
-                .operationIdToResponseAndValidation[operationId];
-
-              return this.failedValidationGuard(existingValidations, groupName);
-            })
-            .map(async (parameterExamples) => {
-              const response = await this.executeRequest(
-                parameterExamples,
-                operationId,
-              );
-              const groupName = Object.keys(parameterExamples)[0];
-
-              if (!this.operationIdToResponseAndValidation[operationId]) {
-                this.operationIdToResponseAndValidation[operationId] = {};
-              }
-
-              this.operationIdToResponseAndValidation[operationId][
-                groupName
-              ] = {
-                response,
+  executeRequests = (): Promise<OperationResponse>[] => {
+    const responses: Promise<OperationResponse>[] = [];
+    for (const operation of this.operations) {
+      for (const exampleGroup of operation.getExampleGroups()) {
+        const operationExampleId = `${operation.getOperationId()}:${exampleGroup.getName()}`;
```

Comment:
This seems to be repeated a few times. Would it make sense to store an array of objects that contain the operationExampleId, operation, and exampleGroup instead of using the nested for loops?

Reply:
this is a great idea!